pkg/gadgets/trace/tcpconnect/types/types.go

```diff
@@ -23,17 +25,25 @@ type Event struct {
 	eventtypes.Event
 	eventtypes.WithMountNsID
 
-	Pid       uint32 `json:"pid,omitempty" column:"pid,template:pid"`
-	Uid       uint32 `json:"uid,omitempty" column:"uid,minWidth:6,hide"`
-	Comm      string `json:"comm,omitempty" column:"comm,template:comm"`
-	IPVersion int    `json:"ipversion,omitempty" column:"ip,width:2,fixed"`
-	Saddr     string `json:"saddr,omitempty" column:"saddr,template:ipaddr"`
-	Daddr     string `json:"daddr,omitempty" column:"daddr,template:ipaddr"`
-	Dport     uint16 `json:"dport,omitempty" column:"dport,template:ipport"`
+	Pid       uint32        `json:"pid,omitempty" column:"pid,template:pid"`
+	Uid       uint32        `json:"uid,omitempty" column:"uid,minWidth:6,hide"`
+	Comm      string        `json:"comm,omitempty" column:"comm,template:comm"`
+	IPVersion int           `json:"ipversion,omitempty" column:"ip,width:2,fixed"`
+	Saddr     string        `json:"saddr,omitempty" column:"saddr,template:ipaddr"`
+	Daddr     string        `json:"daddr,omitempty" column:"daddr,template:ipaddr"`
+	Dport     uint16        `json:"dport,omitempty" column:"dport,template:ipport"`
+	Sport     uint16        `json:"sport,omitempty" column:"sport,template:ipport"`
+	Latency   time.Duration `json:"latency,omitempty" column:"latency,minWidth:8"`
 }
 
 func GetColumns() *columns.Columns[Event] {
-	return columns.MustCreateColumns[Event]()
+	cols := columns.MustCreateColumns[Event]()
+
+	cols.MustSetExtractor("latency", func(event *Event) string {
+		return event.Latency.String()
+	})
```

Comment:
Should the "latency" column be made invisible if the user does not give `--latency`?

Reply:
Absolutely, but I don't know how to make it :grimacing: . We have the same issue in the snapshot process gadget, when the `--threads` parameter is passed, the TID column should be enabled, but it's not. (https://github.com/inspektor-gadget/inspektor-gadget/pull/1396#issuecomment-1489250861)

I tried for a bit to make it work, but it seems it's not straightforward. I opened https://github.com/inspektor-gadget/inspektor-gadget/issues/1578 to track this. Can we move ahead until we have a solution for it?